docs/USAGE-GUIDE.md

```diff
@@ -1585,6 +1585,102 @@ int s2n_config_add_ticket_crypto_key(struct s2n_config *config, const uint8_t *n
+
+To create an external PSK a caller must call the `s2n_external_psk_new` API.**s2n_external_psk_new** creates a new s2n external pre-shared key (PSK) object with `S2N_PSK_HMAC_SHA256` as the default pre-shared key hash algorithm. Note that the obfuscated ticket age will always be zero for external PSKs. Use `s2n_psk_free` to free the memory allocated to the s2n external pre-shared key object created by this API.
+
+**s2n_psk_free** frees the memory associated with the `s2n_psk` object.
+
+After creation of the external PSK object use `s2n_psk_set_identity`, `s2n_psk_set_secret` to set the PSK Identity and PSK secret. To override the default PSK Hash Algorithm  use `s2n_psk_set_hmac`.
+
+**s2n_psk_set_identity** sets the pre-shared key (PSK) identity for a given s2n_psk object. The pre-shared key identity is a unique identifier for the pre-shared secret. The Pre-shared key identity is transmitted over the network unencrypted and is a non-secret value. Be sure to not include any confidential information as a part of the pre-shared key identity. Note that the identity value is copied into s2n-tls memory and the user is responsbile for freeing the memory  associated with the `identity` input. Be sure to call `free(identity)` after this API call. 
+
+**s2n_psk_set_secret** sets the out-of-band/externally provisioned pre-shared secret (PSK) for a given s2n_psk object. Note that the secret value is copied into s2n-tls memory and the user is responsbile for freeing the memory associated with the `secret` input. Be sure to call `free(secret)` after this API call.
+
+**s2n_psk_set_hmac** sets the PSK HMAC algorithm for a given PSK object.
+The supported pre-shared key HMAC algorithms are listed in the `s2n_psk_hmac` enum. Note that in versions >= TLS1.3 the cipher suite selected has the same Hash alogrithm as the pre-shared key. Use this API to override the default PSK Hash Algorithm value `S2N_PSK_HMAC_SHA256`. 
+
+Once the external PSK object is created and updated to include the PSK identity and secret, the caller can add the external PSK to the connection using `s2n_connection_append_psk` API. This API sets up the list of PSKs on the connection that will later be used by the pre-shared key extension to select a pre-shared key for the connection.
+
+**s2n_connection_append_psk** appends a pre-shared key object to the list of pre-shared keys supported by the s2n connection. If a pre-shared key with a duplicate identity is found, an error is returned and the pre-shared key is not added to the list. Note that the PSK object is copied to the connection and the user is responsbile for freeing the memory associated with the s2n_psk object. Use `s2n_psk_free` to free the memory associated with the external pre-shared key object after using this API.
+
+**s2n_config_set_psk_mode** sets the PSK mode on the s2n config object. The enum **s2n_psk_mode** lists the supported PSK modes. 
+
```

Comment:
When would I need to call this API? It's probably worth mentioning in the usage guide that this can be used to switch between external and resumption, but we also do this automatically for the customer in most cases.